src/cli/commands/__tests__/lint.test.ts

```diff
@@ -1,448 +1,172 @@
+  afterEach(() => {
+    errorSpy.mockRestore();
+  });
 
-      test
-        .stdout()
-        .command(['lint', validCustomOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
+  it('shows help when no document argument is passed', async () => {
+    const output = await run('lint');
+    expect(output).toContain('document  Location of a JSON/YAML document');
+  });
 
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain('5:10  warning  info-matches-stoplight  Info must contain Stoplight');
-          expect(ctx.stdout).not.toContain('Info object should contain `contact` object');
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('shows help when invalid arguments are passed', async () => {
+    const output = await run('lint --foo');
+    expect(output).toContain('lint a JSON/YAML document from a file or URL');
+  });
 
-      test
-        .nock('http://foo.local', api =>
-          api.get('/ruleset.yaml').replyWithFile(200, validRulesetPath, {
-            'Content-Type': 'application/yaml',
-          }),
-        )
-        .stdout()
-        .command(['lint', validCustomOas3SpecPath, '-r', 'http://foo.local/ruleset.yaml'])
-        .it('given valid remote ruleset file, outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
-    });
+  it('calls lint with document and default options', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
+        encoding: 'utf8',
+        format: 'stylish',
+      }),
+      undefined,
+    );
+  });
 
-    describe('when a standard oas3 ruleset provided through option', () => {
-      test
-        .stdout()
-        .command(['lint', invalidOas3SpecPath, '-r', standardOas3RulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain(
-            '1:1  warning  api-servers       OpenAPI `servers` must be present and non-empty array',
-          );
-          expect(ctx.stdout).toContain('3:6  warning  info-contact      Info object should contain `contact` object');
-          expect(ctx.stdout).toContain(
-            '3:6  warning  info-description  OpenAPI object info `description` must be present and non-empty string',
-          );
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('calls lint with document and custom encoding', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint --encoding utf16 ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
```

Comment:
ditto

Reply:
In this case, we don't really care about other options - we just want to test whether this one is changed or not.